extensions/resource-deployment/src/ui/deployClusterWizard/deployClusterWizard.ts

```diff
@@ -175,4 +185,44 @@ export class DeployClusterWizard extends WizardBase<DeployClusterWizard, WizardP
 				throw new Error(`Unknown deployment type: ${type}`);
 		}
 	}
+
+	public async refreshPages() {
+		// All the providers will be handled differently
+
+
+
+		const currentPageNumber = this.wizardObject.pages.length;
+
+		for (let i = 1; i < currentPageNumber; i++) {
+			this.wizardObject.removePage(this.wizardObject.pages.length - 1);
+			this.wizardObject.pages.pop();
+		}
+
+		if (instanceOfWizardDeploymentProvider(this.resourceProvider)) {
+			this.wizardInfo = this.resourceProvider.bdcWizard!;
```

Comment:
we should probably be able to do this check in the code that calls refreshPages to make sure we pass in only the correct provider for this wizard.